### PR TITLE
Use OpenRCT2 dialog for saving tracks

### LINKED ISSUE #940: Save Track Dialog could use the same treatment as the Load/Save Dialog.
At current it opens a standard Windows Save Dialog.

### PR DESCRIPTION
**Pull request by SijmenSchoon**
Fixes #940. You might want to test this a bit more extensively, as I've left out a codepath that didn't seem to do anything (when saving returns false).

### CODE REVIEW

**Comment by Wirlie** [comment]
**OS:** Windows 10
 1. Crash after I've clicked on "Use system dialog window".
sin titulo

 2. Almost always I get the error "Track save failed!" when I try to save a track, I'm not sure what is the cause, but, with the latest build of OpenRCT2 I can save a track without any problem.

 3. At some times, I get a crash when I've clicked on the "ok" button.
sin titulo

I'm not sure if this can help you, these are the Dump files:
 1. [5eee6418-ac7d-477c-8cb1-4361a39532ae(034bff3).zip](https://github.com/OpenRCT2/OpenRCT2/files/355425/5eee6418-ac7d-477c-8cb1-4361a39532ae.034bff3.zip)
 2. [c8565543-9a71-4819-8f00-2386de4c17bf(034bff3).zip](https://github.com/OpenRCT2/OpenRCT2/files/355427/c8565543-9a71-4819-8f00-2386de4c17bf.034bff3.zip)

**Comment by SijmenSchoon** [comment]
@Wirlie Fixed, thanks for reporting!

**Comment by Wirlie** [comment]
And, what about the two remaining problems? xP please take a look at that.

**Comment by SijmenSchoon** [comment]
Could someone on Windows please provide me with a stack trace of the second dump? (@Nubbie you, maybe?)

**Comment by SijmenSchoon** [comment]
Fixed @Wirlie's issue 2 as well. I'm having some issues opening the dump file for 3, but once I get that to work (or I manage to reproduce it) that should be fixed soon too.

**Comment by SijmenSchoon** [comment]
@Wirlie Can you try reproducing the issue with a newer (AppVeyor) build? I can't use the dump files because 034bff3 didn't have debugging symbols built :(

File: src/windows/loadsave.c [hunk 1]
```diff
@@ -982,7 +983,7 @@ static char _window_overwrite_prompt_path[MAX_PATH];
 static rct_window *window_overwrite_prompt_open(const char *name, const char *path)
 {
 	rct_window *w;
-
+
```
Comment (LinusU) [inline src/windows/loadsave.c]: added a stray space here

Comment (SijmenSchoon) [inline src/windows/loadsave.c]: Nice spot

**Comment by Wirlie** [comment]
Seems to work fine :P I haven't found any problems.

**Comment by IntelOrca** [comment]
@SijmenSchoon can you now rebase this PR as merging an earlier one has stolen your new strings.

**Comment by SijmenSchoon** [comment]
@IntelOrca Done

**Comment by SijmenSchoon** [comment]
Wait, something went horribly wrong here

**Comment by SijmenSchoon** [comment]
Okay, ready for merging

**Pull request merged**